drivers/modem/modem_socket.c

```diff
@@ -236,7 +237,7 @@ void modem_socket_put(struct modem_socket_config *cfg, int sock_fd)
 
 	k_sem_take(&cfg->sem_lock, K_FOREVER);
 
-	sock->id = cfg->base_socket_num - 1;
+	sock->id = -1;
```

Comment:
I know this is going to sound silly, but I think that in all cases the "not allocated" value should be `cfg->base_socket_num - 1` rather than `-1` outright. I know it seems ridiculous to suggest that a manufacturer would ever use a socket id of negative numbers, but subtracting 1 always guarantees it's out of the range of valid socket ids.